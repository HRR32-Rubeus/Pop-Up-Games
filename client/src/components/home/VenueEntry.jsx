import React from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import WeatherSnip from '../weather/weatherSnip.jsx';

class VenueEntry extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      venue: props.venue,
    };
    this.changeTarget = props.changeTarget;
  }

  render() {
    return (
      <li>
        <div
          className="venuelistentry toneone "
          onClick={() => {
            this.changeTarget({ type: 'venue', id: this.state.venue.id });
            this.props.history.push({
              pathname: 'venue',
              state: {
                lat: this.props.venue.lat,
                lng: this.props.venue.lng,
              },
            });
          }}
        >
          <span className="bold">Name: </span>
          <span>{this.state.venue.venueName}</span>
          <br />
          <br />
          <span className="bold">Address: </span>
          <span>{this.state.venue.address}</span>
          <br />
          <br />
          <WeatherSnip lat={this.props.venue.lat} lng={this.props.venue.lng} />
        </div>
      </li>
    );
  }
}

VenueEntry.propTypes = {
  venue: PropTypes.object.isRequired,
  changeTarget: PropTypes.func.isRequired,
};

export default withRouter(VenueEntry);
