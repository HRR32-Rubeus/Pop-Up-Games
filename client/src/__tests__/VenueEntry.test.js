import VenueEntry from '../components/home/VenueEntry.jsx';
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';

describe('Venue Entry component', () => {
  it('VenueEntry: renders correctly', () => {
    const props = { venue: { name: 'testvenue' }, changeTarget: jest.fn() };
    const tree = renderer
      .create(
        <Router>
          <VenueEntry {...props} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
