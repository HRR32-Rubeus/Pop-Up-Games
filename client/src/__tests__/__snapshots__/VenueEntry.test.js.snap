// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Venue Entry component VenueEntry: renders correctly 1`] = `
<li>
  <div
    className="venuelistentry toneone "
    onClick={[Function]}
  >
    <span
      className="bold"
    >
      Name:
    </span>
    <span />
    <br />
    <br />
    <span
      className="bold"
    >
      Address:
    </span>
    <span />
    <br />
    <br />
  </div>
</li>
`;
